### PR TITLE
Fix various small typos

### PR DESCRIPTION
**Pull request by HeroicEric**
- Fixes some small typos
- Replaces regular dashes (-) with em dashes (—) where appropriate
- Adds some missing periods to the end of bullet items for consistency

### CODE REVIEW

File: src/v0.15/guide/index.md [hunk 1]
```diff
@@ -190,5 +190,5 @@ upfront "wiring" rather than imperative handlers spread throughout your
 codebase.
 
 Although Orbit does not pretend to absorb all the complexity of writing
-ambitious data-driven applications, it's hoped that Orbit's composeable and
+ambitious data-driven applications, it's hoped that Orbit's composable and
```
Comment (HeroicEric) [inline src/v0.15/guide/index.md:193]: /shrug I'm not sure which spelling of this is correct but line 176 uses "composable" so this at least makes them consistent. The layout also uses "composable" https://github.com/HeroicEric/orbitjs-site/blob/620c231d80d5073e73ff22ac4003eac13e3c718e/themes/orbit/layout/index.ejs#L29

Comment (dgeb) [reply src/v0.15/guide/index.md:193]: good catch on the inconsistency - "composable" works for me 👍

File: src/v0.15/guide/data-flows.md [hunk 2]
```diff
@@ -79,7 +80,7 @@ to error as well.
 
 Here are some guidelines for working with data flows:
 
-* Consider the full arc of each request - how it will flow up to be fulfilled,
+* Consider the full arc of each request—how it will flow up to be fulfilled,
```
Comment (dgeb) [inline src/v0.15/guide/data-flows.md:83]: The emdash is an improvement 👍. I will need to verify how it looks without spaces for padding.

Comment (HeroicEric) [reply src/v0.15/guide/data-flows.md:83]: I thought em dashes are supposed to use without spaces around them but I could easily be incorrect :) I just looked again and it seems like it is a matter of preference so LMK if you'd prefer the version with spaces and I can update

## Before

Screen Shot 2019-05-10 at 2 44 02 AM

## After (current)

Screen Shot 2019-05-10 at 2 44 47 AM

## After (with spaces)

Screen Shot 2019-05-10 at 2 49 32 AM

Comment (dgeb) [reply src/v0.15/guide/data-flows.md:83]: Thanks for following up and for the screenshots. After doing some research myself, it does seems that it's quite common to leave out the spaces, so let's go with your current change.

**Comment by dgeb** [comment]
Thanks @HeroicEric  - lots of nice tweaks here 💯

**Pull request merged**